rules/camunda-cloud/escalation-boundary-event-attached-to-ref.js

```diff
@@ -28,7 +28,7 @@ module.exports = skipInNonExecutableProcess(function() {
 
     const attachedToRef = node.get('attachedToRef');
 
-    if (attachedToRef && !isAnyExactly(attachedToRef, [ 'bpmn:CallActivity', 'bpmn:SubProcess' ])) {
+    if (attachedToRef && !isAnyExactly(attachedToRef, [ 'bpmn:CallActivity', 'bpmn:SubProcess', 'bpmn:AdHocSubProcess' ])) {
```

Comment:
Given what you wrote:

> Only Tasks are not allowed to have BoundaryEvents.

We could implement the logic explicitly, couldn't we?

```suggestion
    if (attachedToRef && is(attachedToRef, 'bpmn:Task')) {
```

Reply:
i wanted to keep the change as close as possible to the current implementation, as I was not that familiar with the sideeffects.
For example is a transaction allowed to have boundary events? If no I'll use your suggestion. Currently it is not allowed: 
<img width="1313" alt="image" src="https://github.com/user-attachments/assets/51c215fb-fd3f-4ca2-afde-bbc064d78815" />

Reply:
Transaction is not allowed to be used at all (first problem in the panel), so a diagram with it will always be rejected by the engine.
Let's consider the maintainability of the solution. When/if transaction is implemented, should it be allowed to have a boundary escalation event? We can refer to the issue which triggered the implementation of the rule in the first place: https://github.com/camunda/bpmnlint-plugin-camunda-compat/issues/110
> Escalation boundary events are only supported on elements that spawn a child instance
A transaction, just like any subprocess, will spawn a child instance. Therefore, if the element support is implemented in the engine, we will need to revisit the current rule if we keep the allowed-list solution approach.